src/lib/rc/dsm.cpp

```diff
@@ -108,22 +104,57 @@ dsm_decode(hrt_abstime frame_time, uint16_t *values, uint16_t *num_values, bool
  * @param[out] value pointer to returned channel value
  * @return true=raw value successfully decoded
  */
-static bool
-dsm_decode_channel(uint16_t raw, unsigned shift, unsigned *channel, unsigned *value)
+static bool dsm_decode_channel(uint16_t raw, unsigned shift, uint8_t &channel, uint16_t &value)
 {
-
 	if (raw == 0xffff) {
 		return false;
 	}
 
-	*channel = (raw >> shift) & 0xf;
+	if (shift == 10) {
+		// 1024 Mode: This format is used only by DSM2/22ms mode. All other modes use 2048 data.
+		//  Bits 15-10 Channel ID
+		//  Bits 9-0   Servo Position
+		static constexpr uint16_t MASK_1024_CHANID = 0xFC00;
+		static constexpr uint16_t MASK_1024_SXPOS = 0x03FF;
 
-	uint16_t data_mask = (1 << shift) - 1;
-	*value = raw & data_mask;
+		channel = (raw & MASK_1024_CHANID) >> 10; // 6 bits
+		uint16_t servo_position = (raw & MASK_1024_SXPOS); // 10 bits
+		value = servo_position * 2; // scale to be equivalent to 2048 mode
+		return true;
 
-	//debug("DSM: %d 0x%04x -> %d %d", shift, raw, *channel, *value);
+	} else if (shift == 11) {
+		// 2048 Mode
+		//  Bits 15    Servo Phase
+		//  Bits 14-11 Channel ID
+		//  Bits 10-0  Servo Position
 
```

Comment:
@dagar , here you could mirror line 117 above using what Spektrum calls out in their spec:
```
		static constexpr uint16_t MASK_2048_CHANID = 0x7800;
		static constexpr uint16_t MASK_2048_SXPOS  = 0x07FF ;

		// from Spektrum Remote Receiver Interfacing Rev G Page 6
		uint8_t chan = (raw & MASK_2048_CHANID) >> 11;
		uint16_t servo_position = 0;

		if (chan < 12) {
			// Normal channels
			servo_position = (raw & MASK_2048_SXPOS);
```

Reply:
Done.